afonsoft/VisualStudio
# Plug 1544 - [VisualStudio] for public project if I try to do public scan(1st scan was private). its getting automatically private scan.

**Changes** 
Fixed private-public scan issue

**Test Cases** 

**Test case 1**
- Run a new scan with Project - public and Scan - Public
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - Scan should be Public
- Run a scan with same project - Public 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 2nd Scan should be Public
- Run incremental scan with same project - select public as scan 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 3rd Scan should be public


**Test case 2**
- Run a new scan with Project - public and Scan - Public
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - Scan should be Public
- Run a scan with same project - private 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 2nd Scan should be private
- Run incremental scan with same project - select public as scan 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 3rd Scan should be public


**Test case 3**
- Run a new scan with Project - public and Scan - private
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - Scan should be private
- Run a scan with same project - private 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 2nd Scan should be private
- Run incremental scan with same project - select public as scan 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 3rd Scan should be public


**Test case 4**
- Run a new scan with Project - public and Scan - private
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - Scan should be private
- Run a scan with same project - public 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 2nd Scan should be public
- Run incremental scan with same project - select private as scan 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 3rd Scan should be private

**Test case 5**
- Run a new scan with Project - private and Scan - private (cannot mark it as public scan) 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - Scan should be private
- Run a scan with same project - private (cannot mark it as public scan) 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 2nd Scan should be private
- Run incremental scan with same project - private (By default scan will be private) 
- Wait till scan complete and check it in scan summary (SAST portal)
Expected - 3rd Scan should be private
